server/runtime_javascript_nakama.go

```diff
@@ -250,6 +251,42 @@ func (n *runtimeJavascriptNakamaModule) mappings(r *goja.Runtime) map[string]fun
 		"channelMessageSend":              n.channelMessageSend(r),
 		"channelMessageUpdate":            n.channelMessageUpdate(r),
 		"channeldIdBuild":                 n.channelIdBuild(r),
+		"binaryToString":                  n.binaryToString(r),
+		"stringToBinary":                  n.stringToBinary(r),
+	}
+}
+
+func (n *runtimeJavascriptNakamaModule) binaryToString(r *goja.Runtime) func(goja.FunctionCall) goja.Value {
+	return func(f goja.FunctionCall) goja.Value {
+		if goja.IsUndefined(f.Argument(0)) || goja.IsNull(f.Argument(0)) {
+			panic(r.NewTypeError("expects a Uint8Array object"))
+		}
+
+		data, ok := f.Argument(0).Export().([]byte)
```

Comment:
is it going to be ([]byte) and not https://pkg.go.dev/github.com/dop251/goja#ArrayBuffer ?

Reply:
It looks like it would be wrapped by an `ArrayBuffer` if a typed array object would be created in js, but because we're passing the function a `[]byte` coming from Go directly it is represented as such, without the wrapping.

Reply:
It is an `Array` with integeres elements then, not `ArrayBuffer`. idk where `MatchDataMessage` comes from, assuming it is from the user code sending `Uint8Array`,  and then seeing it in their match loop as JS `Array` will probably be a bit confusing.

It is possible to achieve symmetry by constructing `Uint8Array` from `[]byte` , like following:

```
dataUint8Array, err := vm.New(vm.Get("Uint8Array"), vm.ToValue(vm.NewArrayBuffer(msgData)))
if err != nil {
  panic(err)
}
msgMap["data"] = dataUint8Array
```

Reply:
Yep good point, I've made the change so that we pass a `Uint8Array` to the js runtime.